### ISSUE TITLE
Rename `ec_recover` to `ec_recover_k1`

### ISSUE THREAD

**Issue by Braqzen** [FuelLabs/sway#5219]
With the introduction of #5133 we have some new recovery functionality. 

There are 2 related opcodes, `erc1` & `eck1`. For consistency we should rename `ec_recover` to be explicit similar to the new `r1` function.

**Comment by Braqzen** [comment]
Thoughts cc @bitzoic ?
I'm happy to make the changes.

**Comment by bitzoic** [comment]
No, `erc1` is the default and is used in the Fuel ecosystem. `erk1` should only be used under special circumstances, this was left as is intentionally.

**Comment by Braqzen** [comment]
> No, `erc1` is the default and is used in the Fuel ecosystem. `erk1` should only be used under special circumstances, this was left as is intentionally.

This has nothing to do with defaults. It's about distinguishing between the recovery

**Comment by bitzoic** [comment]
Closing with the introduction of the crypto module https://github.com/FuelLabs/sway/pull/6837
